### PR TITLE
Bodywriting

### PR DESCRIPTION
**Pull request by axietheaxolotl**
## About The Pull Request

I call this one: How I Insert My Fetishes into Skyrat 101

Ports over Body Writing code from Hyper/Vore Code.

Still a WIP, but PRing it for code review.

- [ ] Fix Shit Code
- [ ] Try and make this at least semi-modular
- [ ] Make it so that showering washes it off (Because shower code is completely different here)

## How This Contributes To The Skyrat Roleplay Experience

It's not. But! Drawing dicks on foreheads is funny. Which is a good enough reason.

## Changelog

:cl:
add: Ported Body Writing
/:cl:

### CODE REVIEW

**Comment by Sir-Lagsalot** [comment]
Finally, someone actually putting a decent reason in the "Why it's good for the game" section

File: code/modules/mob/living/carbon/human/examine.dm [hunk 1]
```diff
@@ -322,6 +323,11 @@
 	if(just_sleeping)
 		msg += "[t_He] [t_is]n't responding to anything around [t_him] and seem[p_s()] to be asleep.\n"
 
+	for(var/X in writing) // SKYRAT EDIT ADDITION - BODY WRITING
+		if(!w_uniform)
+			var/obj/item/bodypart/BP = X
+			msg += "<span class='notice'>[capitalize(t_He)] has \"[html_encode(BP.writtentext)]\" written on [t_his] [BP.name].</span>\n"
```
Comment (tf-4) [inline code/modules/mob/living/carbon/human/examine.dm]:
```suggestion
			msg += span_notice("[capitalize(t_He)] has \"[html_encode(BP.writtentext)]\" written on [t_his] [BP.name].\n")
```

File: code/modules/mob/living/carbon/human/examine.dm [hunk 2]
```diff
@@ -322,6 +323,11 @@
 	if(just_sleeping)
 		msg += "[t_He] [t_is]n't responding to anything around [t_him] and seem[p_s()] to be asleep.\n"
 
+	for(var/X in writing) // SKYRAT EDIT ADDITION - BODY WRITING
```
Comment (tf-4) [inline code/modules/mob/living/carbon/human/examine.dm:326]: descriptive var names

File: code/modules/mob/living/carbon/human/examine.dm [hunk 3]
```diff
@@ -322,6 +323,11 @@
...
+		if(!w_uniform)
+			var/obj/item/bodypart/BP = X
```
Comment (tf-4) [inline code/modules/mob/living/carbon/human/examine.dm:328]: descriptive var names

File: code/modules/paperwork/pen.dm [hunk 4]
```diff
@@ -134,14 +134,50 @@
...
+
+			else //writing time
+				var/mob/living/carbon/human/T = M
+				if(!T) //not human.
+					return
+				if(!T.is_chest_exposed())
+					to_chat(user, "<span class='warning'>You cannot write on someone with their clothes on.</span>")
+					return
+
+				var/writting = input(user, "Add writing, doesn't replace current text", "Writing on [T]")  as text|null
```
Comment (tf-4) [inline code/modules/paperwork/pen.dm]:
```suggestion
				var/writing = input(user, "Add writing, doesn't replace current text", "Writing on [target_human]")  as text|null
```

File: code/modules/paperwork/pen.dm [hunk 5]
```diff
@@ -134,14 +134,50 @@
...
+				if(!writting)
```
Comment (tf-4) [inline code/modules/paperwork/pen.dm]:
```suggestion
				if(!writing)
```

File: code/modules/paperwork/pen.dm [hunk 6]
```diff
@@ -134,14 +134,50 @@
...
+					return
+
+				var/obj/item/bodypart/BP = T.get_bodypart(user.zone_selected)
```
Comment (tf-4) [inline code/modules/paperwork/pen.dm]:
```suggestion
				var/obj/item/bodypart/bodypart = target_human.get_bodypart(user.zone_selected)
```

File: code/modules/paperwork/pen.dm [hunk 7]
```diff
@@ -134,14 +134,50 @@
...
+					else
+						to_chat(user, "<span class='notice'>There isnt enough space to write that on [T]'s [BP.name].</span>")
+						return
+
+				if(!(user==T))
+					to_chat(user, "<span class='notice'>You write on [T]'s [BP.name].</span>")
+				else
+					to_chat(user, "<span class='notice'>You write on your [BP.name].</span>")
+	else
+		. = ..()
```
Comment (tf-4) [inline code/modules/paperwork/pen.dm]:
```suggestion
	if(force)
		return ..()
	if(M.can_inject(user, 1))
		if(user.combat_mode) //old poke requires harm intent.
			to_chat(user, span_warning("You stab [M] with the pen."))
			to_chat(M, span_danger("You feel a tiny prick!"))
			log_combat(user, M, "stabbed", src)
			if(!M.try_inject(user, injection_flags = INJECT_TRY_SHOW_ERROR_MESSAGE))
				return FALSE

		else //writing time
			var/mob/living/carbon/human/T = M
			if(!T) //not human.
				return
			if(!T.is_chest_exposed())
				to_chat(user, "<span class='warning'>You cannot write on someone with their clothes on.</span>")
				return

			var/writting = input(user, "Add writing, doesn't replace current text", "Writing on [T]")  as text|null
			if(!writting)
				return

			var/obj/item/bodypart/BP = T.get_bodypart(user.zone_selected)

			if(!(user==T))
				src.visible_message("<span class='notice'>[user] begins to write on [T]'s [BP.name].</span>")
			else
				to_chat(user, "<span class='notice'>You begin to write on your [BP.name].</span>")

			if(do_mob(user, T, 4 SECONDS))
				if((length(BP.writtentext))+(length(writting)) < 100) //100 character limmit to stop spamming.
					BP.writtentext += html_encode(writting) //you can add to text, not remove it.
				else
					to_chat(user, "<span class='notice'>There isnt enough space to write that on [T]'s [BP.name].</span>")
					return

			if(!(user==T))
				to_chat(user, "<span class='notice'>You write on [T]'s [BP.name].</span>")
			else
				to_chat(user, "<span class='notice'>You write on your [BP.name].</span>")
```

File: code/modules/paperwork/pen.dm [hunk 8]
```diff
@@ -134,14 +134,50 @@
...
+		if(M.can_inject(user, 1))
+			if(user.combat_mode) //old poke requires harm intent.
+				to_chat(user, span_warning("You stab [M] with the pen."))
+				to_chat(M, span_danger("You feel a tiny prick!"))
+				log_combat(user, M, "stabbed", src)
+				if(!M.try_inject(user, injection_flags = INJECT_TRY_SHOW_ERROR_MESSAGE))
+					return FALSE
+
+			else //writing time
+				var/mob/living/carbon/human/T = M
```
Comment (tf-4) [inline code/modules/paperwork/pen.dm]:
```suggestion
				var/mob/living/carbon/human/target_human = M
```

File: code/modules/paperwork/pen.dm [hunk 9]
```diff
@@ -134,14 +134,50 @@
...
+				if(!T) //not human.
```
Comment (tf-4) [inline code/modules/paperwork/pen.dm]:
```suggestion
				if(!target_human) //no human.
```

File: code/modules/paperwork/pen.dm [hunk 10]
```diff
@@ -134,14 +134,50 @@
...
+					return
+				if(!T.is_chest_exposed())
```
Comment (tf-4) [inline code/modules/paperwork/pen.dm]:
```suggestion
				if(!get_location_accessible(target_human, BODY_ZONE_CHEST))
```

File: code/modules/paperwork/pen.dm [hunk 11]
```diff
@@ -134,14 +134,50 @@
...
+					to_chat(user, "<span class='warning'>You cannot write on someone with their clothes on.</span>")
```
Comment (tf-4) [inline code/modules/paperwork/pen.dm]:
```suggestion
					to_chat(user, span_warning("You cannot write on someone with their clothes on."))
```

File: code/modules/paperwork/pen.dm [hunk 12]
```diff
@@ -134,14 +134,50 @@
...
+				var/writting = input(user, "Add writing, doesn't replace current text", "Writing on [T]")  as text|null
+				if(!writting)
+					return
+
+				var/obj/item/bodypart/BP = T.get_bodypart(user.zone_selected)
+
+				if(!(user==T))
+					src.visible_message("<span class='notice'>[user] begins to write on [T]'s [BP.name].</span>")
+				else
+					to_chat(user, "<span class='notice'>You begin to write on your [BP.name].</span>")
```
Comment (tf-4) [inline code/modules/paperwork/pen.dm]:
```suggestion
				if(!(user == target_human))
					src.visible_message(span_notice("[user] begins to write on [target_human]'s [bodypart.name]."))
				else
					to_chat(user, span_notice("You begin to write on your [bodypart.name]."))
```

File: code/modules/paperwork/pen.dm [hunk 13]
```diff
@@ -134,14 +134,50 @@
...
+
+				if(do_mob(user, T, 4 SECONDS))
+					if((length(BP.writtentext))+(length(writting)) < 100) //100 character limmit to stop spamming.
+						BP.writtentext += html_encode(writting) //you can add to text, not remove it.
+					else
+						to_chat(user, "<span class='notice'>There isnt enough space to write that on [T]'s [BP.name].</span>")
+						return
```
Comment (tf-4) [inline code/modules/paperwork/pen.dm]:
```suggestion
				if(!(do_mob(user, target_human, 4 SECONDS)))
					return
				if(!((length(bodypart.writtentext))+(length(writing)) < 100)) //100 character limmit to stop spamming.
					to_chat(user, span_notice("There isnt enough space to write that on [target_human]'s [bodypart.name]."))
					return
				bodypart.writtentext += html_encode(writing) //you can add to text, not remove it.
```

File: code/modules/paperwork/pen.dm [hunk 14]
```diff
@@ -134,14 +134,50 @@
...
+
+				if(!(user==T))
+					to_chat(user, "<span class='notice'>You write on [T]'s [BP.name].</span>")
+				else
+					to_chat(user, "<span class='notice'>You write on your [BP.name].</span>")
```
Comment (tf-4) [inline code/modules/paperwork/pen.dm]:
```suggestion
				if(!(user == target_human))
					to_chat(user, span_notice("You write on [target_human]'s [bodypart.name]."))
				else
					to_chat(user, span_notice("You write on your [bodypart.name]."))
```

File: modular_skyrat/modules/__HELPERS/_cit_helpers.dm [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+	return TRUE
+*/
+
+/mob/living/carbon/proc/is_chest_exposed(var/list/L)
+	if(!L)
+		L = get_equipped_items()
+	for(var/obj/item/I in L)
+		if(I.body_parts_covered & CHEST)
+			return FALSE
+	return TRUE
```
Comment (tf-4) [inline modular_skyrat/modules/__HELPERS/_cit_helpers.dm]: don't need any of this
```suggestion
```

**Pull request closed without merging**